### DISCUSSION TITLE
Mirth 3.12.0 There was an error connecting to the server at the specified address. Please verify that the server ip is up and running.

### DISCUSSION THREAD

**Discussion by ppazos** [nextgenhealthcare/connect#5199] [General]
Hi, I just downloaded Mirth Connect 3.12.0 in my Linux Mint 19.3 notebook, which has OpenJDK 8.0.302 installed.

Launched the launcher, and tried to login with admin/admin (it was the first login).

Got "There was an error connecting to the server at the specified address. Please verify that the server ip is up and running."

Netstat shows port 8443 right there, and I was able to log into the web console on that port without problems and with admin/admin.

Then I changed my JDK version to OpenJDK 11.0.10, did the same exact process, and it worked flawlessly, now I can see the Mirth Connect Administrator.

Is there any issue with OpenJDK 8.x that is not documented?

I'm sure JDK 8.x was supported by most of the modern Mirth Connect versions.

Also, I don't know why the error is about a connection, because:

1. the server is up and running, I can confirm that;
2. there is nothing preventing the connection to that port in my machine. I used wireshark to see there is some connection going on since I see the TLS Client Hello/Server Hello/Change Cipher Spec TCP packets.

BTW, mcserver and launcher are running on the same machine.

Any ideas on what is going on?

Thanks!

**Comment by tiskinty** [comment]
I am also seeing this issue happen on a near-daily basis. Running Java 1.8

**Comment by jonbartels** [comment]
What is the exact output of `java -version`? 

Mirth Connect is known to be sensitive to JavaFX, you need a JDK *with JavaFX*. I expect this would be most important for the client but there may be some server side loading or components that also require it.

Anything in `mirth.log`?

Can you post your `mcservice.vmoptions` and `mcserver.vmoptions` please?

When you start MC please use either `./mcservice run` or `./mcserver`. These will run the service in the foreground and may show error messages that fire before the logger starts.

A [documentation search](https://docs.nextgen.com/search?bundle=Mirth_User_Guide_41&q=java%209) in the 4.1.0 user guide turns up:
* [An outdated Java compatibility chart](https://docs.nextgen.com/bundle/Mirth_User_Guide_41/page/connect/connect/topics/c_Java_Requirements_mirth_connect_ug.html)
* [Information on using Java. 9 or newer](https://docs.nextgen.com/bundle/Mirth_User_Guide_41/page/connect/connect/topics/c_Using_Java_9_or_greater_mirth_connect_ug.html), this might be relevant if MC 3.12 flipped from Java 8 as the default to Java 9 as the default.

**Reply by tonygermano** [reply]
While I'm sure that `There was an error connecting to the server at the specified address. Please verify that the server ip is up and running.` can appear when the server is actually not reachable, that is also the error message you get when JavaFX is not available. JavaFX is not actually part of OpenJDK, so you need to either get a java distro that includes it or install a version of OpenJFX that is compatible with your current java version (I'd recommend the former.)
